### ISSUE TITLE
MNIST example get nan loss on mabook pro(intel Iris 655)

### ISSUE THREAD

**Issue by tigerking** [plaidml/plaidml#948]
1. Install plaidml              0.7.0 
2. plaidml-setup to choose "metal_intel(r)_iris(tm)_plus_graphics_655"
3. run sample mnist_cnn.py 
it shows  loss always  "nan"
>>>>>
Using plaidml.keras.backend backend.
x_train shape: (60000, 28, 28, 1)
60000 train samples
10000 test samples
INFO:plaidml:Opening device "metal_intel(r)_iris(tm)_plus_graphics_655.0"
Train on 60000 samples, validate on 10000 samples
Epoch 1/1
 4672/60000 [=>............................] - ETA: 1:02 - loss: nan - acc: 0.0933  
....

Try run  plaidml-setup again to choose "llvm_cpu.0"
run sample mnist_cnn.py  again
the loss appears good now
>>>>>>>
Using plaidml.keras.backend backend.
x_train shape: (60000, 28, 28, 1)
60000 train samples
10000 test samples
INFO:plaidml:Opening device "llvm_cpu.0"
Train on 60000 samples, validate on 10000 samples
Epoch 1/1
  128/60000 [..............................] - ETA: 1:17:12 - loss: 2.3035 - acc: 0.1641

Can anyone give some hints to figure the issue? Thanks

**Comment by tigerking** [comment]
it looks that lots of guys met this issue but no solution yet.  
"some guys said that update batch size from 128 to 127 or 129 , the loss:nan disappear but training result  is very bad  (VS using vanilla back end Keras).

**Comment by robsalasco** [comment]
I have he same issue trying to train an example neural network from keras

**Comment by vlada-rozova** [comment]
The same exact problem with Intel Iris Graphics 550 1536 MB, has anyone solved this?

**Comment by LindqU** [comment]
> it looks that lots of guys met this issue but no solution yet.
> "some guys said that update batch size from 128 to 127 or 129 , the loss:nan disappear but training result is very bad (VS using vanilla back end Keras).
> 
> "some guys said, it works if using AMD GPU (15'' MAC with internal Intel Iris GPU and AMD CPU), ,but met the same issue, lost:nan, using Iris 6xx GPU.

FYI
this  issues is being discussed in [issues 168](https://github.com/plaidml/plaidml/issues/168)

[isuues 168] is more Active.
Maybe someone here can resoluson you out.
